File: Matlab/SimpleBlockadePlot.m
%Script for plotting blockade shift results
%24/07/2017

close all
clear
clc

atom = '87Rb';
nn = 50;
ll = 0;
jj = 0.5;
mj = 0.5;

[RRSI,theta,blockadeshiftGHzmesh,C_6val] =...
    BlockadeShift(atom,nn,ll,jj,mj);

stringlookup = 'SPD';
[numer, denom] = rat(jj);
[numer1, denom1] = rat(mj);

figure
mesh(RRSI*1e6,theta,blockadeshiftGHzmesh*1e3);
xlabel('$R\, (\mu\mathrm{m})$','interpreter','latex');
ylabel('$\theta\, (\mathrm{radians})$','interpreter','latex');
zlabel('$\Delta\!W\, (\mathrm{MHz})$','interpreter','latex');
set(gcf,'Color','w');
axis tight;
title(['Calculated Rydberg blockade shift for |', num2str(nn),...
    stringlookup(ll+1), '_{', num2str(numer), '/', num2str(denom),...
    '}, m_j = ', num2str(numer1), '/', num2str(denom1),...
    '\rangle state of Rb']);

File: Python/SimpleBlockadePlot.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Sat May 18 16:27:43 2019

@author: Craig Chisholm
"""

#This script is just the blockade shift part of Demonstrations.py

import matplotlib.pyplot as plt
import numpy as np
import RydbergFunctions as Rydberg
from fractions import Fraction
from matplotlib import cm

plt.close("all")

#Input information
atom = '87Rb'
nn = 50
ll = 2
jj = 2.5
mj = 2.5

RRSI, theta, blockadeshiftGHzmesh, C_6val = Rydberg.BlockadeShift(atom,nn,ll,jj,mj)

stringlookup = 'SPD'
ratJ = Fraction(jj)
numer = ratJ.numerator
denom = ratJ.denominator
ratmJ = Fraction(mj)
numer1 = ratmJ.numerator
denom1 = ratmJ.denominator

plt.figure()
plt.pcolormesh(np.multiply(RRSI,1e6),theta,np.multiply(blockadeshiftGHzmesh,1e3),cmap=cm.inferno,shading='auto')
plt.xlabel('$R\, (\mu\mathrm{m})$')
plt.ylabel(r'$\theta\, (\mathrm{radians})$')
plt.title(f'Calculated Rydberg blockade shift for\n $|{nn}{stringlookup[ll]}_{{{numer}/{denom}}}, m_j = {numer1}/{denom1}\\rangle$ state of {atom}')
plt.colorbar()
plt.show()